metrics_api/test/opentelemetry/metrics/meter_test.rb

```diff
@@ -24,11 +24,21 @@
       _(counter.class).must_equal(OpenTelemetry::Metrics::Instrument::Counter)
     end
 
+    it 'test create_counter with advisory parameters' do
```

Comment:
Considering tests as documentation, I think it would be helpful to add assertions for the other instruments with their permitted advisory parameters.